android/tangram/src/com/mapzen/tangram/FontFileParser.java

```diff
@@ -0,0 +1,138 @@
+package com.mapzen.tangram;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+import org.xmlpull.v1.XmlPullParserFactory;
+
+import android.util.Xml;
+import android.util.Log;
+
+class FontFileParser {
+
+    private Map<String, String> fontDict = new HashMap<String, String>();
+
+    private void processDocument(XmlPullParser parser) throws XmlPullParserException, IOException {
+
+        parser.nextTag();
+        // Parse Families
+        parser.require(XmlPullParser.START_TAG, null, "familyset");
+        while (parser.next() != XmlPullParser.END_DOCUMENT) {
+            if (parser.getEventType() != XmlPullParser.START_TAG) {
+                continue;
+            }
+            if (parser.getName().equals("family")) {
```

Comment:
If `parser.getName()` returns null this will throw a NullPointerException

Better to invert the order and place the string literal first.

```java
if ("family".equals(parser.getName()) { ... }
```